src/hosted-form/iframe-content/hosted-input-validator.ts

```diff
@@ -0,0 +1,77 @@
+import { cvv, expirationDate, number } from 'card-validator';
+import { pick } from 'lodash';
+import { object, string, StringSchema, ValidationError } from 'yup';
+
+import HostedInputValidateErrorData from './hosted-input-validate-error-data';
+import HostedInputValues from './hosted-input-values';
+
+interface HostedInputValidateOptions {
+    isCardCodeRequired?: boolean;
+}
+
+export default class HostedInputValidator {
+    async validate(values: HostedInputValues, options?: HostedInputValidateOptions): Promise<HostedInputValidateErrorData[]> {
+        const config: { [key in keyof HostedInputValues]: StringSchema } = {
+            cardExpiry: this._getCardExpirySchema(),
+            cardName: this._getCardNameSchema(),
+            cardNumber: this._getCardNumberSchema(),
+        };
+
+        if (options && options.isCardCodeRequired) {
+            config.cardCode = this._getCardCodeSchema();
+        }
+
+        try {
+            await object(pick(config, Object.keys(values)))
+                .validate(values, { abortEarly: false });
```

Comment:
could you use `validateSync` for this?, to avoid using await.

Reply:
Prefer to keep it as `validate` if that's alright with you. Using `async/await` essentially makes the code read as if I'm using `validateSync`.